scripts/install_gtsam.sh

```diff
@@ -3,30 +3,21 @@
 #--- Script for installing GTSAM ---#
 
 
-### Default imported variables setup ###
-
-if [ -z "${opt_dir}" ]; then
-    opt_dir="/opt/ILLIXR"
-fi
-
-if [ -z "${illixr_nproc}" ]; then
-    illixr_nproc="1"
-fi
+### Setup ###
 
+## Source the default values for imported variables
+. scripts/default_values.sh
 
-### Helper functions ###
-
-# Source the global helper functions
-. scripts/bash_utils.sh
+## Source the global helper functions
+. scripts/system_utils.sh
```

Comment:
Why isn't this being done for all the other install scripts (openxr, qemu, etc.)? They also use the variables sourced in these two bash files, and to further add to the confusion, a subset of these variables is also set in `install_deps.sh` (e.g., line 82 sets `opt_dir`).

Reply:
> Why isn't this being done for all the other install scripts (openxr, qemu, etc.)?

Other scripts will be moved to a similar structure in dedicated PR's for those installs (to reduce the time to merge for this PR, as well as to improve separation of concerns for debugging). In the future, these scripts should be replaced for portability (e.g., moving dependency handling to the python/Runner), and should target the cross-platform package manager resulting from #104 and #246.

> They also use the variables sourced in these two bash files, and to further add to the confusion, a subset of these variables is also set in install_deps.sh (e.g., line 82 sets opt_dir).

The default values can be overridden by calling scripts like `install_deps.sh`. There are three possible workflows to consider:

1.  Calling script is `Dockerfile`: The values set in the environment take precedent when debugging/testing in CI.

1.  Calling script is `install_deps.sh`: This interactive script sets the values for a local machine install. We don't take user input in this script for the installation locations yet, but the defaults can be overridden here.

1.  There is no calling script (the script is called directly from the CLI or another script): The script being called will not crash due to missing (or erroneously empty) values, and use the default install setup can continue.

I agree these scripts should be cleaned up to unify the structure and reduce confusion. I don't think we should address those issues in this PR.